### PR TITLE
Add analytics event for Custom Domains.

### PR DESCRIPTION
**Pull request by jklausa**
Adds basic analytics for Custom Domains.

`-Shared` counterpart PR is: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/168

To test:

Look at the code, and like, trust me that it works ;P
Optionally, just go trough the whole flow and verify that appropriate events are printed to the console.

### CODE REVIEW

**Comment by Terri5732** [comment]
Where are the other WP apps like BuddyPress

**Comment by Terri5732** [comment]
Bbstyle

File: Podfile [hunk 1]
```diff
@@ -24,13 +24,13 @@ end
...
-    pod 'WordPressShared', '~> 1.5.0'
+    ##pod 'WordPressShared', '~> 1.5.0'
 
     ## for development:
-    ## pod 'WordPressShared', :path => '../WordPress-iOS-Shared'
+    ##pod 'WordPressShared', :path => '../WordPress-iOS-Shared'
 
     ## while PR is in review:
-    ## pod 'WordPressShared', :git => 'https://github.com/wordpress-mobile/WordPress-iOS-Shared.git', :commit => '14a5ebae28fdc09130b091330d84a3a310132cd2'
+    pod 'WordPressShared', :git => 'https://github.com/wordpress-mobile/WordPress-iOS-Shared.git', :commit => 'd664be9e496112b8f2d98f41607db41cbcc6fd0b'
```
Comment (jkmassel) [inline Podfile]: @jklausa – just tagged 1.5.1, so please replace the hash with that before merging! ❤️

Comment (jklausa) [reply Podfile]: Donezo!

**Comment by jklausa** [comment]
That's what the `.lock` file is there for though ;P? But sure, no problem. Updated!

**Comment by jkmassel** [comment]
> That's what the `.lock` file is there for though ;P? But sure, no problem. Updated!

Yah I'm just suuuuper paranoid of someone (me, maybe?) running `pod update` when they meant to do `pod install` and something changing. Or missing the tiny change in the `.lock` file in a big diff.

My philosophy: if you're going to update a dependency, _you gotta really want it_ 🙃

Thank you for indulging my paranoia 😅

**Pull request merged**